app/src/main/java/ar/scacchipa/twittercloneapp/viewmodel/LoginApiViewModel.kt

```diff
@@ -0,0 +1,49 @@
+package ar.scacchipa.twittercloneapp.viewmodel
+
+import android.net.Uri
+import androidx.core.net.toUri
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import ar.scacchipa.twittercloneapp.datasource.UserAccessToken
+import ar.scacchipa.twittercloneapp.datasource.provideAuthSourceDateApi
+import ar.scacchipa.twittercloneapp.datasource.provideRetrofit
+import ar.scacchipa.twittercloneapp.domain.AuthorizationUseCase
+import ar.scacchipa.twittercloneapp.repository.AuthorizationRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class LoginApiViewModel (
+    private val userCase: AuthorizationUseCase =
+        AuthorizationUseCase(AuthorizationRepository(provideAuthSourceDateApi(provideRetrofit())))
+): ViewModel() {
+    var userAccessToken = MutableLiveData<UserAccessToken>()
+        private set
```

Comment:
making it private set means that the variable cannot change what it is pointing to, but not means that the clients of this class can put things inside that mutable live data. Remember to always share to others non mutable versions of livedata.

Reply:
this was replaced with

```
private val _userAccessToken = MutableLiveData<UserAccessToken>()
    val userAccessToken: LiveData<UserAccessToken>
        get() = _userAccessToken
```